### ISSUE TITLE
Where is Inductive codes for GraphSAGE?

### ISSUE THREAD

**Issue by HarrialX** [dmlc/dgl#1408]
Hi all, a simple question about [GraphSAGE samples](https://github.com/dmlc/dgl/tree/master/examples/pytorch/graphsage), I run the train_full.py but seems like that one is transductive learning. Any inductive graph learning sample codes about GraphSAGE?

**Comment by BarclayII** [comment]
Hi,

Sorry for the extremely late reply.  There is currently no existing code for inductive learning with GraphSAGE.  However, the only difference between inductive learning and transductive learning (in terms of definition in GraphSAGE's paper) is that the training graph and the testing graph are different: the training graph only consists of the nodes in the training and validation set, while the testing graph also includes the test set nodes.

One can simply train the GraphSAGE model on the subgraph by replacing the graph with `g.subgraph(torch.cat([train_node_ids, val_node_ids]))`.

Please feel free to follow up.
Thanks.

**Comment by snash4** [comment]
Hi Barclayll, 
 I didn't understand your answer, could you please explain it more, thanks.

In inductive learning, the objective is to get an embedding of an unseen node(a new node in graph) from a trained model. By this definition, the model doesn't need to be re-trained. Correct me if I am wrong.

**Comment by HongHuangNeu** [comment]
@BarclayII  Correct me if I am wrong:
What you do to deal with unseen node is: 
(1)add the initial embedding of the unseen nodes to the node tensor
(2)add edges intrioduced by these unseen nodes  to the original trainging graph
(3)feed the new node tensor and new graph to the model and get the resulting tensor. The vector at the corresponding position of node tensor should be the embedding vector for the unseen nodes

Am I correct?
